src/goto-symex/build_goto_trace.cpp

```diff
@@ -326,8 +393,8 @@ void build_goto_trace(
       for(goto_tracet::stepst::iterator
           s_it2=s_it1;
           s_it2!=goto_trace.steps.end();
-          s_it2=goto_trace.steps.erase(s_it2));
-
+          s_it2=goto_trace.steps.erase(s_it2))
+      { };
```

Comment:
As you are touching this code: I believe this can be rewritten to `goto_trace.steps.erase(s_it1, goto_trace.steps.end());` instead of this strange for loop.